storage/innobase/include/fts0types.ic

```diff
@@ -79,19 +79,18 @@ fts_ranking_doc_id_cmp(
 }
 
 /******************************************************************//**
-Compare two fts_update_t doc_ids.
+Compare two doc_ids.
 @return < 0 if n1 < n2, 0 if n1 == n2, > 0 if n1 > n2 */
 UNIV_INLINE
-int
-fts_update_doc_id_cmp(
+int doc_id_cmp(
 /*==================*/
 	const void*	p1,			/*!< in: id1 */
 	const void*	p2)			/*!< in: id2 */
 {
-	const fts_update_t*	up1 = (const fts_update_t*) p1;
-	const fts_update_t*	up2 = (const fts_update_t*) p2;
+	const doc_id_t*	up1 = (const doc_id_t*) p1;
+	const doc_id_t*	up2 = (const doc_id_t*) p2;
 
-	return((int)(up1->doc_id - up2->doc_id));
+	return((int)(up1 - up2));
```

Comment:
And here `static_cast<>` too.

Reply:
There was a bigger problem, actually :)

Reply:
Good that it was caught! But apparently, it wasn't covered by tests.